ksqldb-engine/src/main/java/io/confluent/ksql/util/ValidationSharedKafkaStreamsRuntimeImpl.java

```diff
@@ -15,192 +15,103 @@
-  private final Map<QueryId, Set<SourceName>> sources;
-  private final KafkaStreamsBuilder kafkaStreamsBuilder;
+public class ValidationSharedKafkaStreamsRuntimeImpl extends SharedKafkaStreamsRuntime {
+  private final Logger log = LoggerFactory.getLogger(ValidationSharedKafkaStreamsRuntimeImpl.class);
 
   public ValidationSharedKafkaStreamsRuntimeImpl(
-      final SharedKafkaStreamsRuntime sharedKafkaStreamsRuntime
+      final SharedKafkaStreamsRuntime sharedRuntime
   ) {
-    streamsProperties = new ConcurrentHashMap<>(sharedKafkaStreamsRuntime.getStreamProperties());
-    streamsProperties.put(StreamsConfig.APPLICATION_ID_CONFIG,
-        "_confluent-ksql-" + UUID.randomUUID() + "-validation"
+    super(
+        sharedRuntime.getKafkaStreamsBuilder(),
+        getSandboxStreamsProperties(sharedRuntime),
+        new QueryMetadataImpl.TimeBoundedQueue(Duration.ofHours(1), 0)
     );
-    kafkaStreamsBuilder = sharedKafkaStreamsRuntime.getKafkaStreamsBuilder();
-    kafkaStreams = kafkaStreamsBuilder.buildNamedTopologyWrapper(streamsProperties);
-    metadata = new ConcurrentHashMap<>(sharedKafkaStreamsRuntime.getMetadata());
-    sources = new ConcurrentHashMap<>(sharedKafkaStreamsRuntime.getSourcesMap());
-    for (PersistentQueriesInSharedRuntimesImpl queryMetadata : metadata.values()) {
+
+    for (BinPackedPersistentQueryMetadata query : sharedRuntime.collocatedQueries.values()) {
       //kafkaStreams.addNamedTopology(queryMetadata.getTopology());
     }
   }
 
-  public ValidationSharedKafkaStreamsRuntimeImpl(final KafkaStreamsBuilder kafkaStreamsBuilder,
-                                                 final int maxQueryErrorsQueueSize,
-                                                 final Map<String, Object> streamsProperties) {
-    kafkaStreams = kafkaStreamsBuilder.buildNamedTopologyWrapper(streamsProperties);
-    this.streamsProperties = ImmutableMap.copyOf(streamsProperties);
-    metadata = new ConcurrentHashMap<>();
-    sources = new ConcurrentHashMap<>();
-    this.kafkaStreamsBuilder = kafkaStreamsBuilder;
+  public ValidationSharedKafkaStreamsRuntimeImpl(
+      final KafkaStreamsBuilder kafkaStreamsBuilder,
+      final int maxQueryErrorsQueueSize,
+      final Map<String, Object> streamsProperties
+  ) {
+    super(
+        kafkaStreamsBuilder,
+        streamsProperties,
+        new QueryMetadataImpl.TimeBoundedQueue(Duration.ofHours(1), maxQueryErrorsQueueSize)
+    );
   }
 
-  public void markSources(final QueryId queryId, final Set<SourceName> sourceNames) {
-    sources.put(queryId, sourceNames);
+  private static Map<String, Object> getSandboxStreamsProperties(
+      final SharedKafkaStreamsRuntime sharedKafkaStreamsRuntime
+  ) {
+    final Map<String, Object> sandboxStreamsProperties =
+        new ConcurrentHashMap<>(sharedKafkaStreamsRuntime.getStreamProperties());
+    sandboxStreamsProperties.put(
+        StreamsConfig.APPLICATION_ID_CONFIG,
```

Comment:
We fixed the format that we make appIds for the normal runtimes here https://github.com/confluentinc/ksql/blob/03af146beb1cc68bea95779c8360b70f10708b01/ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryBuilder.java#L504

Can you make this match? it was for metrics so it shouldn't be a huge deal but I would rather have the runtime Ids match as much as possible

Reply:
Will do. Actually this right here should just be tacking on the `"-validation"` suffix to the application id of the original/real, but thanks for reminding/pointing me to that --  I noticed there's a `QueryApplicationId` builder class already so I factored the shared runtimes id logic into a separate builder method in that class so we can keep all the application id logic together. See [here ](https://github.com/confluentinc/ksql/pull/8355/files#diff-fd36d70d052543249b4a009833353071168d34ea1631d26d49f32d8f41abce41R28-R40)